src/collide_vss.cpp

```diff
@@ -550,17 +566,149 @@ void CollideVSS::EEXCHANGE_NonReactingEDisposal(Particle::OnePart *ip,
         }
         postcoln.evib += p->evib;
       } // end of vibdof if
+      if (species[sp].elecdat != NULL) {
+        int *estates = particle->eivec[particle->ewhich[index_elecstate]];
+        double elec_phi = get_elec_phi(p->ispecies, jp->ispecies, estates[p - particle->particles], E_Dispose);
+        if (elec_phi >= random->uniform()) {
+          relax_electronic_mode(p, jp, E_Dispose);
+        }
+      }
     }
   }
 
+  // compute post-collision internal energies
+
+  postcoln.erot = ip->erot + jp->erot;
+  postcoln.evib = ip->evib + jp->evib;
+  postcoln.eelec = 0.0;
+  if (index_eelec >= 0) {
+    double *eelecs = particle->edvec[particle->ewhich[index_eelec]];
+    if (species[ip->ispecies].elecdat != NULL)
+      postcoln.eelec += eelecs[ip - particle->particles];
+    if (species[jp->ispecies].elecdat != NULL)
+      postcoln.eelec += eelecs[jp - particle->particles];
+  }
+
   // compute portion of energy left over for scattering
 
-  postcoln.eint = postcoln.erot + postcoln.evib;
+  postcoln.eint = postcoln.erot + postcoln.evib + postcoln.eelec;
   postcoln.etrans = E_Dispose;
 }
 
 /* ---------------------------------------------------------------------- */
 
+void CollideVSS::relax_electronic_mode(Particle::OnePart *p, Particle::OnePart *jp, double& E_Dispose)
+{
+  Particle::Species *species = particle->species;
+  double *eelecs = particle->edvec[particle->ewhich[index_eelec]];
+  E_Dispose += eelecs[p - particle->particles];
+
+  int ielec = select_elec_state(
+    p, jp, E_Dispose,
+    params[p->ispecies][jp->ispecies].omega,
+    species[p->ispecies].elecdat->enforce_spin_conservation[jp->ispecies] );
+  double eelec = species[p->ispecies].elecdat->states[ielec].temp*update->boltz;
+
+  eelecs[p - particle->particles] = eelec;
+
+  int *estates = particle->eivec[particle->ewhich[index_elecstate]];
+  estates[p - particle->particles] = ielec;
+  E_Dispose -= eelecs[p - particle->particles];
+}
+
+/* ---------------------------------------------------------------------- */
+
+double CollideVSS::get_elec_phi(int ispec1, int ispec2, int ielec, double)
+{
+  if (particle->species[ispec1].elecdat->species_rel[ispec2] != NULL) {
+    return particle->species[ispec1].elecdat->species_rel[ispec2][ielec];
+  } else {
+    return particle->species[ispec1].elecdat->default_rel[ielec];
+  }
+}
+
+/* ---------------------------------------------------------------------- */
+
+double CollideVSS::calc_elec_coll_temp(Particle::OnePart *p, double E_Dispose, double omega)
```

Comment:
@zseckert this function `CollideVSS::calc_elec_coll_temp` doesn't seem to be used, can it be removed?

Reply:
Removed. It was originally intended for people who would want to implement temperature dependent relaxation models for electronic states, but the electronic temperature is very non-intuitive, so making them think hard about this before undertaking such a task is probably a good thing.